### PR TITLE
Add io.github.Figma_Linux.figma_linux

### PR DESCRIPTION
**Pull request by jannuary**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build io.github.Figma_Linux.figma_linux

**Comment by jannuary** [comment]
bot, build io.github.Figma_Linux.figma_linux

**Comment by jannuary** [comment]
bot, build io.github.Figma_Linux.figma_linux

**Comment by jannuary** [comment]
bot, build io.github.Figma_Linux.figma_linux

File: io.github.Figma_Linux.figma_linux.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        <content_attribute id="social-contacts">intense</content_attribute>
+        <content_attribute id="money-purchasing">intense</content_attribute>
+    </content_rating>
+    <releases>
+        
+
+/&gt;
+        
+        
+
```
Comment (barthalion) [inline io.github.Figma_Linux.figma_linux.appdata.xml:42]: Looks like some leftover?

File: io.github.Figma_Linux.figma_linux.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+runtime-version: '20.08'
+base: org.electronjs.Electron2.BaseApp
+base-version: '20.08'
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --device=dri
+  - --share=network
+  - --filesystem=home
+  - --filesystem=xdg-pictures
```
Comment (barthalion) [inline io.github.Figma_Linux.figma_linux.yml:13]: `home` includes `xdg-pictures`.

File: io.github.Figma_Linux.figma_linux.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      - cp -r  opt/figma-linux /app/figma-linux
+      - mkdir -p /app/share/icons/hicolor
+      - cp -r usr/share/icons/hicolor/* /app/share/icons/hicolor
+      - install figma-linux.sh /app/bin/figma-linux
+      - install -Dm644 io.github.Figma_Linux.figma_linux.desktop /app/share/applications/io.github.Figma_Linux.figma_linux.desktop
+      - install -Dm644 ${FLATPAK_ID}.appdata.xml -t /app/share/appdata/
+    sources:
+      - type: file
+        only-arches:
+          - x86_64
```
Comment (barthalion) [inline io.github.Figma_Linux.figma_linux.yml:33]: This is not technically needed as you also added flathub.json.

File: io.github.Figma_Linux.figma_linux.desktop [hunk 4]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Name=figma-linux
+Comment=Unofficial desktop application for linux
+Exec=/app/bin/figma-linux --no-sandbox --enable-oop-rasterization --ignore-gpu-blacklist -enable-experimental-canvas-features --enable-accelerated-2d-canvas --force-gpu-rasterization --enable-fast-unload --enable-accelerated-vpx-decode=3 --enable-tcp-fastopen --javascript-harmony --enable-checker-imaging --v8-cache-options=code --v8-cache-strategies-for-cache-storage=aggressive --enable-zero-copy --ui-enable-zero-copy --enable-native-gpu-memory-buffers --enable-webgl-image-chromium --enable-accelerated-video --enable-gpu-rasterization %U
```
Comment (barthalion) [inline io.github.Figma_Linux.figma_linux.desktop:4]:
```suggestion
Exec=figma-linux --enable-oop-rasterization --ignore-gpu-blacklist -enable-experimental-canvas-features --enable-accelerated-2d-canvas --force-gpu-rasterization --enable-fast-unload --enable-accelerated-vpx-decode=3 --enable-tcp-fastopen --javascript-harmony --enable-checker-imaging --v8-cache-options=code --v8-cache-strategies-for-cache-storage=aggressive --enable-zero-copy --ui-enable-zero-copy --enable-native-gpu-memory-buffers --enable-webgl-image-chromium --enable-accelerated-video --enable-gpu-rasterization %U
```

File: io.github.Figma_Linux.figma_linux.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+        url: https://github.com/Figma-Linux/figma-linux/releases/download/v0.7.2/figma-linux_0.7.2_linux_amd64.deb
+        sha256: e3fef97591d716701f400064ac085308437a259a2395cb74337cccd6ef606dd9
+        size: 76809218
+
+        x-checker-data:
+          type: debian-repo
+          package-name: figma-linux
+          dist: all
+          component: main
```
Comment (barthalion) [inline io.github.Figma_Linux.figma_linux.yml:43]: There's no debian-repo here so it won't work. Using JSON checker with GitHub API might be a better idea: https://github.com/flathub/flatpak-external-data-checker/#json-checker

**Review by barthalion — COMMENTED** [review]
The desktop and appdata files should be ideally upstreamed. Just to double-check, @ChugunovRoman, would you be interested in co-maintaining your app on Flathub?

**Comment by nedrichards** [comment]
bot, build io.github.Figma_Linux.figma_linux

File: flathub.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: upstream has an aarch64 build. any reason to exclude it here?

File: io.github.Figma_Linux.figma_linux.yml (hunk 5 above)
Comment (ChugunovRoman) [reply io.github.Figma_Linux.figma_linux.yml:42]: > The desktop and appdata files should be ideally upstreamed. Just to double-check, @ChugunovRoman, would you be interested in co-maintaining your app on Flathub?

Ok, I don't mind the package support on Flathub.
Am I correct in understanding that after accepting this PR, a repository will be created and I will just need to update it?

File: flathub.json (hunk 6 above)
Comment (ChugunovRoman) [reply flathub.json:2]: Ok, I think need to add.
How to determine the arch of a package which need download in the source section in the YAML file?
I mean the link: `https://github.com/Figma-Linux/figma-linux/releases/download/v0.7.2/figma-linux_0.7.2_linux_amd64.deb`

Comment (barthalion) [reply flathub.json:2]: You can define `only-arches` also in the specific source file (along with checksum etc).

File: io.github.Figma_Linux.figma_linux.yml (hunk 5 above)
Comment (nedrichards) [reply io.github.Figma_Linux.figma_linux.yml:42]: Yes, that's exactly it.

**Pull request closed without merging**